rcgen/src/string_types.rs

```diff
@@ -0,0 +1,317 @@
+///
+/// For the full ASCII character set, use
+/// [`Ia5String`][`crate::Ia5String`].
+///
+/// # Supported characters
+///
+/// The following ASCII characters/ranges are supported:
+///
+/// - `A..Z`
+/// - `a..z`
+/// - `0..9`
+/// - "` `" (i.e. space)
+/// - `\`
+/// - `(`
+/// - `)`
+/// - `+`
+/// - `,`
+/// - `-`
+/// - `.`
+/// - `/`
+/// - `:`
+/// - `=`
+/// - `?`
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct PrintableString(String);
+
+impl PrintableString {
+	/// Extracts a string slice containing the entire `PrintableString`.
+	pub fn as_str(&self) -> &str {
+		&self.0
+	}
+}
+
+impl TryFrom<&str> for PrintableString {
+	type Error = Error;
+
+	fn try_from(input: &str) -> Result<Self, Error> {
+		input.to_string().try_into()
+	}
+}
+
+impl TryFrom<String> for PrintableString {
+	type Error = Error;
+
+	fn try_from(value: String) -> Result<Self, Self::Error> {
+		for &c in value.as_bytes() {
+			match c {
+				b'A'..=b'Z'
+				| b'a'..=b'z'
+				| b'0'..=b'9'
+				| b' '
+				| b'\''
+				| b'('
+				| b')'
+				| b'+'
+				| b','
+				| b'-'
+				| b'.'
+				| b'/'
+				| b':'
+				| b'='
+				| b'?' => (),
+				_ => return Err(Error::InvalidPrintableString),
+			}
+		}
+		Ok(Self(value))
+	}
+}
+
+impl FromStr for PrintableString {
+	type Err = Error;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		s.try_into()
+	}
+}
+
+/// ASN.1 `IA5String` type.
+///
+/// Supports the [International Alphabet No. 5 (IA5)] character encoding, i.e.
+/// the lower 128 characters of the ASCII alphabet. (Note: IA5 is now
+/// technically known as the International Reference Alphabet or IRA as
+/// specified in the ITU-T's T.50 recommendation).
+///
+/// For UTF-8, use [`String`][`std::string::String`].
+///
+/// [International Alphabet No. 5 (IA5)]: https://en.wikipedia.org/wiki/T.50_%28standard%2
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Ia5String(String);
+
+impl Ia5String {
+	/// Extracts a string slice containing the entire `Ia5String`.
+	pub fn as_str(&self) -> &str {
+		&self.0
+	}
+}
+
+impl TryFrom<&str> for Ia5String {
+	type Error = Error;
+
+	fn try_from(input: &str) -> Result<Self, Error> {
+		input.to_string().try_into()
+	}
+}
+
+impl TryFrom<String> for Ia5String {
+	type Error = Error;
+
+	fn try_from(input: String) -> Result<Self, Error> {
+		// Validate all characters are within IA5String's allowed set
+		if input.as_bytes().iter().any(|&c| c > 0x7F) {
+			return Err(Error::InvalidPrintableString);
+		}
+		Ok(Self(input.to_string()))
+	}
+}
+
+impl FromStr for Ia5String {
+	type Err = Error;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		s.try_into()
+	}
+}
+
+/// ASN.1 `TeletexString` type.
+///
+/// # Supported characters
+///
+/// The standard defines a complex character set allowed in this type. However, quoting the ASN.1
+/// mailing list, "a sizable volume of software in the world treats TeletexString (T61String) as a
```

Comment:
Do you have a reference for the ASN.1 mailing list post? Seems worth linking as a footnote.

Reply:
I added it it the new commit ;)